src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -692,6 +692,10 @@ def status(self):
         subvolume_status = {'state' : state.value}
 
         if subvolume_type == SubvolumeTypes.TYPE_CLONE:
+            if SubvolumeOpSm.is_retained_state(state):
```

Comment:
This has been added to ensure that `fs clone status` reports error for subvolumes that were created through cloning but now have been deleted with `--retain-snapshots` option.

Reply:
Is this trying to workaround stale rstats that never gets updated?

Reply:
No, it's just making sure that `fs clone status` doesn't run successfully for subvolume deleted with `--retain-snapshots`.

Reply:
Any why would that be required?

Reply:
We have a test(s) in test_volume.py that checks this. It will fail without this.

Reply:
I'm not following. Is this a bug you have found (unrelated to this change)?

Reply:
No.

Reply:
We want to be able to open subvolumes which have been deleted with `--retain-snapshots`. If we allow that, "ceph fs clone status"  will run successfully even for such subvolumes. This leads to failures among tests.

Thus we need to add a check (which the check above) to `status()` that would prevent "ceph fs clone status" from running successfully in these cases.

Reply:
Had a conversation with @rishabh-d-dave - this bug got introduced in the PR. So, please link the commit which introduced this bug. Otherwise I will have to go over the entire change over again.

Reply:
Given yesterday's update to this PR, this change is redundant and therefore has been removed.

Reply:
Sorry, how is this redundant? You did mention that one of the previous commits introduced the bug. So, which change made this redundant?

Reply:
Yeah, but in recent changes that has been fixed. Test result provide extra confirmation on this.